test/ReverseProxy.Tests/Transforms/ResponseHeaderValueTransformTests.cs

```diff
@@ -12,35 +12,38 @@ public class ResponseHeaderValueTransformTests
-        [InlineData("start", 200, "new", false, true, "new")]
-        [InlineData("start", 400, "new", true, false, "start")]
-        [InlineData("start", 200, "new", true, false, "start;new")]
-        [InlineData("start", 400, "new", true, true, "start;new")]
-        [InlineData("start", 200, "new", true, true, "start;new")]
-        [InlineData("start,value", 400, "new", true, false, "start,value")]
-        [InlineData("start,value", 200, "new", true, false, "start,value;new")]
-        [InlineData("start,value", 400, "new", true, true, "start,value;new")]
-        [InlineData("start,value", 200, "new", true, true, "start,value;new")]
-        [InlineData("start;value", 400, "new", true, false, "start;value")]
-        [InlineData("start;value", 200, "new", true, false, "start;value;new")]
-        [InlineData("start;value", 400, "new", true, true, "start;value;new")]
-        [InlineData("start;value", 200, "new", true, true, "start;value;new")]
-        public async Task AddResponseHeader_Success(string startValue, int status, string value, bool append, bool always, string expected)
+        [InlineData("", 400, "new", false, false, "", false)]
+        [InlineData("", 502, "new", false, false, "", true)]
+        [InlineData("", 200, "new", false, false, "new", false)]
+        [InlineData("", 400, "new", false, true, "new", false)]
+        [InlineData("", 200, "new", false, true, "new", false)]
+        [InlineData("", 502, "new", false, true, "new", true)]
+        [InlineData("start", 400, "new", false, false, "start", false)]
+        [InlineData("start", 200, "new", false, false, "new", false)]
+        [InlineData("start", 502, "new", false, false, "start", true)]
+        [InlineData("start", 400, "new", false, true, "new", false)]
+        [InlineData("start", 200, "new", false, true, "new", false)]
+        [InlineData("start", 400, "new", true, false, "start", false)]
+        [InlineData("start", 200, "new", true, false, "start;new", false)]
+        [InlineData("start", 400, "new", true, true, "start;new", false)]
+        [InlineData("start", 200, "new", true, true, "start;new", false)]
+        [InlineData("start,value", 400, "new", true, false, "start,value", false)]
+        [InlineData("start,value", 200, "new", true, false, "start,value;new", false)]
+        [InlineData("start,value", 400, "new", true, true, "start,value;new", false)]
+        [InlineData("start,value", 200, "new", true, true, "start,value;new", false)]
+        [InlineData("start;value", 400, "new", true, false, "start;value", false)]
+        [InlineData("start;value", 200, "new", true, false, "start;value;new", false)]
+        [InlineData("start;value", 400, "new", true, true, "start;value;new", false)]
+        [InlineData("start;value", 200, "new", true, true, "start;value;new", false)]
+        public async Task AddResponseHeader_Success(string startValue, int status, string value, bool append, bool always, string expected, bool responseNull)
```

Comment:
```suggestion
        public async Task AddResponseHeader_Success(string startValue, int status, string value, bool append, bool always, string expected, bool responseNull = false)
```
Nit: May be more readable since only a handful of cases set it.